lightning/src/ln/msgs.rs

```diff
@@ -742,6 +743,26 @@ impl Readable for OptionalField<Script> {
 	}
 }
 
+impl Writeable for OptionalField<u64> {
+	fn write<W: Writer>(&self, w: &mut W) -> Result<(), ::std::io::Error> {
+		match *self {
+			OptionalField::Present(ref value) => {
+				value.write(w)?;
+			},
+			OptionalField::Absent => {}
+		}
+		Ok(())
+	}
+}
+
+impl Readable for OptionalField<u64> {
+	fn read<R: Read>(r: &mut R) -> Result<Self, DecodeError> {
+		let value: u64 = Readable::read(r)?;
```

Comment:
I think this is why fuzzing fails... Well, at least I see why fuzzer *might* fail here.

`Serialized(Deserialized(UnsignedChannelAnnouncement1)) != UnsignedChannelAnnouncement1` if htlc_maximum_msat is present, but the flag is not set (or vice versa). Because the flag logic is one level above these serializers.
And fuzzers are probably not aware of this logic?
Not sure how to make it work properly.